test/test_nn.py

```diff
@@ -4305,6 +4305,93 @@ def test_conv_double_backward_cuda(self):
                                 "\ninp_size: " + str(inp_size) +
                                 "\ndilation: " + str(dilation))
 
+    def test_grad_conv2d_weight(self):
```

Comment:
Is there a way to not duplicate the code here? For example, you are also going to need 1d as well